token/js/test/e2e-2022/tokenMetadata.test.ts

```diff
@@ -0,0 +1,421 @@
+            connection,
+            payer,
+            authority.publicKey,
+            mint.publicKey,
+            authority,
+            'name',
+            'symbol',
+            'uri',
+            undefined,
+            undefined,
+            TEST_PROGRAM_ID
+        );
+    });
+
+    it('can successfully initialize', async () => {
+        const meta = await getTokenMetadata(connection, mint.publicKey, undefined, TEST_PROGRAM_ID);
+        expect(meta).to.deep.equal({
+            updateAuthority: new PublicKey('ExgT3gCWXJzY4a9SHqTqsTk6dPAj37WNq2uWNbmMG1JR'),
+            mint: mint.publicKey,
+            name: 'name',
+            symbol: 'symbol',
+            uri: 'uri',
+            additionalMetadata: [],
+        });
+    });
+
+    it('can successfully emit', async () => {
+        const signature = await tokenMetadataEmit(connection, payer, mint.publicKey);
+
+        // Get metadata emitted in the transaction to verify
+        const tx: any = await connection.getTransaction(signature, {
+            maxSupportedTransactionVersion: 2,
+        });
+
+        const data = Buffer.from(tx?.meta?.returnData?.data?.[0], 'base64');
+
+        expect(unpack(data)).to.deep.equal({
+            updateAuthority: new PublicKey('ExgT3gCWXJzY4a9SHqTqsTk6dPAj37WNq2uWNbmMG1JR'),
+            mint: mint.publicKey,
+            name: 'name',
+            symbol: 'symbol',
+            uri: 'uri',
+            additionalMetadata: [],
+        });
+    });
+
+    it('can successfully update when reducing', async () => {
+        await tokenMetadataUpdateField(
+            connection,
+            payer,
+            authority,
+            mint.publicKey,
+            'symbol',
+            'TEST',
+            undefined,
+            undefined,
+            TEST_PROGRAM_ID
+        );
+
+        const meta = await getTokenMetadata(connection, mint.publicKey, undefined, TEST_PROGRAM_ID);
+
+        expect(meta).to.deep.equal({
+            updateAuthority: new PublicKey('ExgT3gCWXJzY4a9SHqTqsTk6dPAj37WNq2uWNbmMG1JR'),
+            mint: mint.publicKey,
+            name: 'name',
+            symbol: 'TEST',
+            uri: 'uri',
+            additionalMetadata: [],
+        });
+    });
+
+    it('can successfully update with rent transfer', async () => {
+        await Promise.all([
+            tokenMetadataUpdateFieldWithRentTransfer(
+                connection,
+                payer,
+                authority,
+                mint.publicKey,
+                'TVL',
+                '1,000,000',
+                undefined,
+                undefined,
+                TEST_PROGRAM_ID
+            ),
+            tokenMetadataUpdateFieldWithRentTransfer(
+                connection,
+                payer,
+                authority,
+                mint.publicKey,
+                'name',
+                'TEST',
+                undefined,
+                undefined,
+                TEST_PROGRAM_ID
+            ),
+        ]);
```

Comment:
Yes that is correct - mostly did it this way to make the tests run faster and I am checking the metadata once both complete so don't think the order should matter. 

But to make things easier I split it into sequential operations. I didn't split into separate tests to to avoid running the beforeEach hook an additional time